### PR TITLE
Validate checkoutline quantity

### PR DESCRIPTION
**Pull request by d-wysocki**
I want to merge this change because...I want to validate checkout line quantity for update and create lines.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2881 CheckoutLinesAdd and CheckoutCreate shouldn't accept quantity=0 in inputs.](https://app.clickup.com/t/2549495/SALEOR-2881)

File: saleor/graphql/checkout/tests/test_checkout.py [hunk 1]
```diff
@@ -1517,6 +1542,26 @@ def test_checkout_lines_add_with_unavailable_variant(
     assert errors[0]["variants"] == [variant_id]
 
 
+def test_checkout_lines_add_with_zero_quantity(
```
Comment (fowczarek) [inline saleor/graphql/checkout/tests/test_checkout.py:1545]: Missing test for `CheckoutLinesUpdate`  mutation.

Comment (d-wysocki) [reply saleor/graphql/checkout/tests/test_checkout.py:1545]: Which test is missing?
For `CheckoutLinesUpdate` witch zero quantity we already had test - `test_checkout_line_delete_by_zero_quantity`

File: saleor/graphql/checkout/mutations.py [hunk 2]
```diff
@@ -113,10 +113,11 @@ def update_checkout_shipping_method_if_invalid(
         checkout.save(update_fields=["shipping_method", "last_change"])
 
 
-def check_lines_quantity(variants, quantities, country, channel_slug):
+def check_lines_quantity(variants, quantities, country, channel_slug, new_line=True):
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations.py:116]: The flag parameter is a bad idea, but if we must use it here, I would rename it to something like `allow_zero_quantity=False`. In the docstring, I would add: "By default, zero quantity is not allowed, but if this validation is used for updating existing checkout lines, `allow_zero_quantities` can be set to `True` and checkout lines with this quantity can be later removed."

**Review by maarcingebala — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Pull request merged**